examples/azure/terraform-azure-availabilityset-example/README.md

```diff
@@ -0,0 +1,109 @@
+1. Install [Terraform](https://www.terraform.io/) and make sure it's on your `PATH`.
+1. [Review environment variables](#review-environment-variables).
+1. Install [Golang](https://golang.org/) and make sure this code is checked out into your `GOPATH`.
+1. `cd test`
+1. Make sure [the azure-sdk-for-go versions match](#check-go-dependencies) in [/go.mod](/go.mod) and in [test/azure/terraform_azure_availabilityset_example_test.go](/test/azure/terraform_azure_availabilityset_example_test.go).
+1. `go build terraform_azure_availabilityset_example_test.go`
+1. `go test -v -run TestTerraformAzureAvailabilitySetExample`
+
+## Module test APIs
+
+This module exposes the following top-level API methods:
+
+- `AvailabilitySetExists` indicates whether the speficied Azure Availability Set  exists\
+    func AvailabilitySetExists(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) bool
+
+- `GetAvailabilitySetFaultDomainCount` gets the Fault Domain Count for the specified Azure Availability Set\
+    func GetAvailabilitySetFaultDomainCount(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) int32
+
+- `CheckAvailabilitySetContainsVM` checks if the Virtual Machine is contained in the Availability Set VMs\
+    func CheckAvailabilitySetContainsVM(t testing.TestingT, vmName string, avsName string, resGroupName string, subscriptionID string) bool
+
+- `GetAvailabilitySetVMs` gets a list of VM names in the specified Azure Availability Set\
+    func GetAvailabilitySetVMs(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) []string
+    
+- `GetAvailabilitySetE` gets an Availability Set in the specified Azure Resource Group\
+    func GetAvailabilitySetE(t testing.TestingT, avsName string, resGroupName string, subscriptionID string) (*compute.AvailabilitySet, error)
+    
+- `GetAvailabilitySetClientE` gets a new Availability Set client in the specified Azure Subscription\
+    func GetAvailabilitySetClientE(subscriptionID string) (*compute.AvailabilitySetsClient, error)
+
+## Check Go Dependencies
+
+Check that the `github.com/Azure/azure-sdk-for-go` version in your generated `go.mod` for this test matches the version in the terratest [go.mod](https://github.com/gruntwork-io/terratest/blob/master/go.mod) file.  
+
+> This was tested with **go1.14.4**.
+
+### Check Azure-sdk-for-go version
+
+Let's make sure [go.mod](https://github.com/gruntwork-io/terratest/blob/master/go.mod) includes the appropriate [azure-sdk-for-go version](https://github.com/Azure/azure-sdk-for-go/releases/tag/v46.1.0):
+
+```go
+require (
+    ...
+    github.com/Azure/azure-sdk-for-go v46.1.0+incompatible
+    ...
+)
+```
+
+If we make changes to either the **go.mod** or the **go test file**, we should make sure that the go build command works still.
+
+```powershell
+go build terraform_azure_availabilityset_test.go
+```
+
+## Review Environment Variables
```

Comment:
I think we were discussing this in Slack, but it may make sense to extract this into, say, `examples/azure/README.md` and have all the individual examples cross-link to it rather than copy/pasting it, as this is a good amount of content.

Reply:
Moved general instructions to examples/azure/README.md and linked to it in step 4.